openmdao/core/group.py

```diff
@@ -864,13 +864,17 @@ def get_relevance_graph(self, desvars, responses):
                     missing_responses.add(output)
 
         if missing_responses:
-            msg = (f"Constraints or objectives [{', '.join(sorted(missing_responses))}] cannot"
-                   " be impacted by the design variables of the problem because no partials "
-                   "were defined for them in their parent component(s).")
-            if self._problem_meta['singular_jac_behavior'] == 'error':
-                raise RuntimeError(msg)
+            for subsys in self.system_iter(include_self=True, recurse=True, typ=Group):
```

Comment:
Rather than skipping the entire check, we could just modify the `_get_missing_partials` method in Group and Component to just not report any missing partials within that system (or below in the case of Group).  That way, the dependency graph will still leave those system nodes in the graph (so they'll be treated as 'dense', i.e. all outputs depend on all inputs) for those nodes.